src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static int CAPACITY = 10;
+    private static double SIZE_OF_INCREASE = 1.5;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[CAPACITY];
+    }
+
+    private void increaseCapacity() {
+        if (values.length == size) {
+            int newCapacity = (int)(values.length * SIZE_OF_INCREASE);
+            values = Arrays.copyOf(values, newCapacity);
+        }
+    }
+
+    private void correctIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        increaseCapacity();
+        values[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
```

Comment:
The index validation logic here is repeated and should be extracted into the 'correctIndex' method. However, since 'add' allows adding an element at the end of the list (index == size), the 'correctIndex' method should be adjusted to allow this index as well.